virttest/utils_net.py

```diff
@@ -4468,3 +4468,88 @@ def delete_linux_bridge_tmux(linux_bridge_name, iface_name=None, ignore_status=F
     else:
         cmd = 'ip link delete %s' % linux_bridge_name
     return utils_misc.cmd_status_output(cmd, shell=True, verbose=True, ignore_status=ignore_status)
+
+
+def check_filter_rules(ifname, bandwidth, expect_none=False):
+    """
+    Check bandwidth settings via 'tc filter' output
+
+    :param ifname: the interface name
+    :param bandwidth: the settings in the xml
+    :param expect_none: whether or not expect nothing in output,
+                        default to be False
+    :return: return True if get expected result, else return False
+    """
+    cmd = "tc -d filter show dev %s parent ffff:" % ifname
+    filter_output = process.run(cmd, shell=True).stdout_text
+    logging.debug("Bandwidth filter output: %s", filter_output)
+    if expect_none:
+        return not filter_output.strip()
+    if not filter_output.count("filter protocol all pref"):
+        logging.error("Can't find 'protocol all' settings in filter rules")
+        return False
+    filter_pattern = r".*police.*rate (\d+)(K?M?)bit burst (\d+)(K?M?)b.*"
+    se = re.search(r"%s" % filter_pattern, filter_output, re.M)
+    if not se:
+        logging.error("Can't find any filter policy")
+        return False
+    logging.debug("bandwidth from tc output:%s" % str(se.groups()))
+    logging.debug("bandwidth from setting:%s" % str(bandwidth))
+    if "average" in bandwidth:
+        if se.group(2) == 'M':
+            tc_average = int(se.group(1)) * 1000
+        else:
+            tc_average = int(se.group(1))
+        assert tc_average == int(bandwidth["average"]) * 8
```

Comment:
Do we need to catch exception for any odd cases?

Reply:
Yes, the bandwidth includes 3 values, average, peak and burst, each value should be compared with the outputs from tc.

Reply:
Oops, I think you misunderstood my comment.
My question is if you need to add L4498~4509 under 'try' block and try to catch up AssertionError. I'm asking that because you're going to make this func returns True or False.

Reply:
Updated, please help to check, Thank you!